crates/admin/src/storage_query/mod.rs

```diff
@@ -8,20 +8,17 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use restate_storage_query_datafusion::context::QueryContext;
+mod error;
+mod query;
 
-/// Handlers share this state
-#[derive(Clone)]
-pub struct EndpointState {
-    query_context: QueryContext,
-}
+use axum::{routing::post, Router};
+use std::sync::Arc;
 
-impl EndpointState {
-    pub fn new(query_context: QueryContext) -> Self {
-        Self { query_context }
-    }
+use crate::state::QueryServiceState;
 
-    pub fn query_context(&self) -> &QueryContext {
-        &self.query_context
-    }
+pub fn create_router(state: Arc<QueryServiceState>) -> Router<()> {
+    // Setup the router
+    axum::Router::new()
+        .route("/query", post(query::query))
```

Comment:
This change will probably also require another fix for the dashboard if it relied on the `api` prefix.